test/unit/application_validator_test.rb

```diff
@@ -24,29 +24,6 @@ class ApplicationValidatorTest < Minitest::Test
       assert result.ok?, result.error_value
     end
 
-    # TODO: add test in the context of the extension template to ensure that extensions work as expected
-
-    test "check_acyclic_graph returns error when package set contains circular dependencies" do
-      use_template(:minimal)
-      merge_into_app_yaml_file("components/sales/package.yml", { "dependencies" => ["components/timeline"] })
-      merge_into_app_yaml_file("components/timeline/package.yml", { "dependencies" => ["components/sales"] })
-
-      result = validator.check_acyclic_graph(package_set)
-
-      refute result.ok?
-      assert_match(/Expected the package dependency graph to be acyclic/, result.error_value)
-      assert_match %r{components/sales → components/timeline → components/sales}, result.error_value
-    end
-
-    test "check_acyclic_graph returns allows nonexistent packages" do
```

Comment:
If we're grouping acyclic and nonexistent package checks under the same validator I think that makes sense. I just wanted to make sure this didn't regress on the specific method since it was a bug, and I thought one day we could break up the app validator into separate ones (like you're doing now).